test/integration/bulk-action-integration.ts

```diff
@@ -0,0 +1,202 @@
+  })
+
+  after(async () => {
+    if (writeSpace) {
+      return writeSpace.delete()
+    }
+  })
+
+  describe('Read', () => {
+    test('Get BulkAction', async () => {
+      const entry = await testEnvironment.getEntry(TestDefaults.entry.testEntryId)
+
+      const createdBulkAction = await testEnvironment.createPublishBulkAction(
+        bulkActionPayload([makeVersionedLink('Entry', entry.sys.id, entry.sys.version)])
+      )
+
+      const bulkActionInProgress = await testEnvironment.getBulkAction(createdBulkAction.sys.id)
+      expect(bulkActionInProgress.sys.id).to.eql(createdBulkAction.sys.id)
+    })
+
+    test('Get BulkAction on wrong id', async () => {
+      try {
+        await testEnvironment.getBulkAction('fakeId')
+      } catch (error) {
+        const parsed = JSON.parse(error.message)
+        expect(parsed.status).to.eql(404)
+        expect(parsed.message).to.eql('The resource could not be found.')
+        expect(parsed.details).to.eql({
+          type: 'BulkAction',
+          id: 'fakeId',
+        })
+      }
+    })
+  })
+
+  describe('Write', () => {
+    test('Publish BulkAction', async () => {
+      const entry = await testEnvironment.getEntry(TestDefaults.entry.testEntryId)
+
+      const createdBulkAction = await testEnvironment.createPublishBulkAction(
+        bulkActionPayload([makeVersionedLink('Entry', entry.sys.id, entry.sys.version)])
+      )
+
+      // Wait for BulkAction completion
+      await delay(1000)
+
+      const bulkActionInProgress = await testEnvironment.getBulkAction(createdBulkAction.sys.id)
+      expect(bulkActionInProgress.sys.status).to.eql(BulkActionStatus.succeeded)
+      expect(bulkActionInProgress.action).to.eql('publish')
+    })
+
+    test('Publish BulkAction with wrong payload', async () => {
+      const entry = await testEnvironment.getEntry(TestDefaults.entry.testEntryId)
+
+      // The publish action relies on the Link object having a `version` property
+      try {
+        await testEnvironment.createPublishBulkAction(
+          bulkActionPayload([makeLink('Entry', entry.sys.id)])
+        )
+      } catch (error) {
+        const parsed = JSON.parse(error.message)
+        expect(parsed.status).to.eql(422)
+        expect(parsed.message).to.eql('Validation error')
+        expect(parsed.details).to.eql({
+          errors: [
+            {
+              details: '"entities.items[0].sys.version" is required',
+              name: 'any.required',
+              path: ['entities', 'items', 0, 'sys', 'version'],
+            },
+          ],
+        })
+      }
+    })
+
+    test('Unpublish BulkAction', async () => {
+      const entry = await testEnvironment.getEntry(TestDefaults.entry.testEntryId)
+
+      const createdBulkAction = await testEnvironment.createUnpublishBulkAction(
+        bulkActionPayload([makeLink('Entry', entry.sys.id)])
+      )
+
+      // Wait for BulkAction completion
+      await delay(1000)
```

Comment:
Is it save to assume the bulk action is done after a second? I think this might become flaky.
Can't we poll the api until the bulk action is done?

Reply:
I was thinking about it, do we have a "poller" in the SDK for these kind of entities? @andipaetzold 

I was planning to add a `waitForCompletion` kind of function (in `entities/bulk-action.ts`) that handles it. Let me know if it's a good idea.

Reply:
I don't think we currently have such a thing in the sdk. Adding it would make sense as it's probably a common use case to send the request and just wait for it to complete.

Reply:
@andipaetzold I added a new method for the bulkAction itself (using the normal client) that waits for the processing to be done.

For the plain API, I didn't find anything that could be used except exposing a helper function